test/e2e/seeder/anvil.ts

```diff
@@ -0,0 +1,149 @@
+  host: '127.0.0.1',
+  mnemonic:
+    'spread raise short crane omit tent fringe mandate neglect detail suspect cradle',
+  port: 8545,
+};
+
+export class Anvil {
+  #server: AnvilType | undefined;
+
+  async start(opts = defaultOptions): Promise<void> {
+    const options = { ...defaultOptions, ...opts };
+
+    // Determine the path to the anvil binary directory
+    const anvilBinaryDir = join(process.cwd(), 'node_modules', '.bin');
+
+    // Prepend the anvil binary directory to the PATH environment variable
+    process.env.PATH = `${anvilBinaryDir}:${process.env.PATH}`;
+
+    // Verify that the anvil binary is accessible
+    try {
+      const versionOutput = execSync('anvil --version', { encoding: 'utf-8' });
+      console.log(`Anvil version: ${versionOutput}`);
+      console.log(
+        `Anvil server started on port: ${options.port} with chainId: ${options.chainId}`,
+      );
+    } catch (error) {
+      console.error('Failed to execute anvil:', error);
+      throw new Error('Anvil binary is not accessible.');
+    }
+
+    this.#server = createAnvil(options);
+    await this.#server.start();
+  }
+
+  getProvider() {
+    if (!this.#server) {
+      throw new Error('Server not running yet');
+    }
+    const { walletClient, publicClient, testClient } = createAnvilClients(
+      this.#server,
+      this.#server.options.chainId ?? 1337,
+      this.#server.options.port ?? 8545,
+    );
+
+    return { walletClient, publicClient, testClient };
+  }
+
+  async getAccounts(): Promise<string[]> {
+    const provider = this.getProvider();
+
+    const { walletClient } = provider;
+    const accounts = await walletClient.getAddresses();
+    return accounts;
+  }
+
+  async getBalance(address: `0x${string}` | null = null): Promise<number> {
+    const provider = this.getProvider();
+
+    if (!provider) {
+      console.log('No provider found');
+      return 0;
+    }
+    const { publicClient } = provider;
+
+    const accountToUse = address || (await this.getAccounts())?.[0];
+
+    if (!accountToUse) {
+      console.log('No accounts found');
+      return 0;
+    }
+
+    const balanceInt = await publicClient.getBalance({
+      address: accountToUse as `0x${string}`,
+    });
+    const balanceFormatted = Number(balanceInt) / 10 ** 18;
+
+    // Round to four decimal places, so we return the same value as ganache does
+    const balanceRounded = parseFloat(balanceFormatted.toFixed(4));
+    return balanceRounded;
+  }
+
+  async getFiatBalance(): Promise<number> {
+    const balance = await this.getBalance();
+    const currencyConversionRate = 1700.0;
+    const fiatBalance = (balance * currencyConversionRate).toFixed(2);
+
+    return Number(fiatBalance);
+  }
+
+  async setAccountBalance(
+    address: `0x${string}`,
+    balance: string,
+  ): Promise<void> {
+    const provider = this.getProvider();
+    const { testClient } = provider;
+
+    if (!provider) {
+      throw new Error('No provider found');
```

Comment:
Is this line ever reachable?

Reply:
good point, yeah I think we wouldn't get there if !provider, it was copied from the Ganache class. I've removed it as unnecessary